yarn-project/pxe/src/tagging/sync/sync_sender_tagging_indexes.ts

```diff
@@ -0,0 +1,121 @@
+import type { AztecAddress } from '@aztec/stdlib/aztec-address';
+import type { AztecNode } from '@aztec/stdlib/interfaces/server';
+import type { DirectionalAppTaggingSecret } from '@aztec/stdlib/logs';
+import { TxStatus } from '@aztec/stdlib/tx';
+
+import type { SenderTaggingDataProvider } from '../../storage/tagging_data_provider/sender_tagging_data_provider.js';
+import { loadAndStoreNewTaggingIndexes } from './load_and_store_new_tagging_indexes.js';
+
+// This window has to be as large as the largest expected number of logs emitted in a tx for a given directional app
+// tagging secret. If we get more tag indexes consumed than this window, an error is thrown in `PXE::proveTx` function.
+// This is set to a larger value than MAX_PRIVATE_LOGS_PER_TX (currently 32) because there could be more than
+// MAX_PRIVATE_LOGS_PER_TX indexes consumed in case the logs are squashed. This happens when the log contains a note
+// and the note is nullified in the same tx.
```

Comment:
I don't understand what this comment is trying to say. Does it mean we cannot find a correct length?